packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -42,23 +41,24 @@ interface SideNavGroupProps {
   </SideNavItem>
 </SideNavGroup>
  * ```
+ *
  * @param props.className Class name that will be applied to the root-level element.
- * @param props.headerText Text that will be rendered as the component's header.
- * @param props.headerClassName Class name that will be applied to the component's header.
+ * @param props.header Content that will be rendered as the component's header
+ *   If a string is provided, it will be rendered with default styling as a header tag.
  * @param props.children Class name that will be applied to the component's header.
- *
  */
-function SideNavGroup({
-  headerText,
-  headerClassName,
-  children,
-  ...rest
-}: SideNavGroupProps) {
+function SideNavGroup({ header, children, ...rest }: SideNavGroupProps) {
+  const renderHeader = () => {
```

Comment:
micro opt, rather than incurring a new function call cost we could do:

```jsx
const renderedHeader = typeof header === 'string' ? <h4 className={cx(sideNavLabelStyle)}>{header}</h4> : header;
```

Could even inline it.